fileglancer/proxiedpath.py

```diff
@@ -0,0 +1,96 @@
+import requests
+import logging
+
+from typing import Optional
+from functools import cache
+
+from .uimodels import ProxiedPath, CachedEntry
+
+
+log = logging.getLogger(__name__)
+
+
+class ProxiedPathManager:
+    """Manage the list of user shared data paths from the central server."""
+
+    def __init__(self, central_url: str):
+        self.central_url = central_url
+        self._cached_proxied_paths = {}
+    
+    def get_proxied_paths(self, username: str, sharing_key: Optional[str] = None) -> list[ProxiedPath]:
+        """Lookup a user shared path by its sharing key."""
+        cached_user_proxied_paths = self._cached_proxied_paths.get(username)
+        if cached_user_proxied_paths is not None and cached_user_proxied_paths.is_not_expired():
+            user_proxied_paths = cached_user_proxied_paths.entry()
+        else:
+            user_proxied_paths = self._cache_user_proxied_paths(username).entry()
+        if user_proxied_paths is None:
+            return None
+        elif sharing_key is not None:
+            return [p for p in user_proxied_paths if p.sharing_key == sharing_key]
+        else:
+            return user_proxied_paths
+
+    def create_proxied_path(self, username: str, mount_path: str) -> requests.Response:
+        """Create a proxied path with <a_path> as the mount_point"""
+        del self._cached_proxied_paths[username]  # invalidate the cache
+        return requests.post(
+            f"{self.central_url}/proxied-path/{username}",
+            params = {
+                "mount_path": mount_path
+            }
+        )
+
+    def delete_proxied_path(self, username: str, sharing_key: str) -> requests.Response:
+        del self._cached_proxied_paths[username]  # invalidate the cache
+        return requests.delete(
+            f"{self.central_url}/proxied-path/{username}/{sharing_key}"
+        )
+
+    def update_proxied_path(self, username: str, sharing_key, new_path: Optional[str], new_name: Optional[str]) -> requests.Response:
+        """Update a proxied path with <a_path> as the mount_point"""
+        del self._cached_proxied_paths[username]  # invalidate the cache
+        pp_updates = {}
+        if new_path:
+            pp_updates["mount_path"] = new_path
+        if new_name:
+            pp_updates["sharing_name"] = new_name
+        return requests.put(
+            f"{self.central_url}/proxied-path/{username}/{sharing_key}",
+            params = pp_updates
+        )
+
+    def _cache_user_proxied_paths(self, username: str) -> list[ProxiedPath] | None:
+        """Retrieve and cache user ."""
+        if self.central_url and username:
+            log.info(f"Cache proxied paths for user {username} from central server")
+            response = requests.get(f"{self.central_url}/proxied-path/{username}")
+            response.raise_for_status()
+            pps_json = response.json()["paths"]
```

Comment:
@krokicki I think the central API will require a change for this to return the proxied paths just like shared paths are returned.

Reply:
Yeah it currently returns a list, it's probably better to always return an object.